packages/payment-processor/src/payment/erc777-stream.ts

```diff
@@ -0,0 +1,55 @@
+import { ContractTransaction, Signer } from 'ethers';
+
+import { ClientTypes, ExtensionTypes, PaymentTypes } from '@requestnetwork/types';
+
+import {
+  getPaymentNetworkExtension,
+  getProvider,
+  getRequestPaymentValues,
+  validateRequest,
+} from './utils';
+import { Framework } from '@superfluid-finance/sdk-core';
+
+/**
+ * Processes a transaction to pay an ERC777 stream Request.
+ * @param request
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param _overrides optionally, override default transaction values, like gas.
+ */
+export async function payErc777StreamRequest(
+  request: ClientTypes.IRequestData,
+  signer: Signer,
+): Promise<ContractTransaction> {
+  const id = getPaymentNetworkExtension(request)?.id;
+  if (id !== ExtensionTypes.ID.PAYMENT_NETWORK_ERC777_STREAM) {
+    throw new Error('Not a supported ERC777 payment network request');
+  }
+  validateRequest(request, PaymentTypes.PAYMENT_NETWORK_ID.ERC777_STREAM);
+  const networkName =
+    request.currencyInfo.network === 'private' ? 'custom' : request.currencyInfo.network;
+  // console.log('networkName:', networkName);
+
+  const sf = await Framework.create({
+    networkName: networkName,
+    provider: signer.provider ? signer.provider : getProvider(),
+    dataMode: 'WEB3_ONLY',
+    resolverAddress: '0x8e4C131B37383E431B9cd0635D3cF9f3F628EDae', //this is how you get the resolver address
```

Comment:
> //this is how you get the resolver address

What do you mean by this, as the resolver address is written as plain text?

Reply:
I think this comes from the documentation. @bertux can you adapt the comment so it's clear from a Request context please?

Reply:
@yomarion you guessed it well, it was an environment variable described by this comment in the documentation.
I've adapted by adding the test to use this address only for local blockchain. For public blockchains it's resolved automatically by the SF SDK.